lib/mongo/cluster.rb

```diff
@@ -214,6 +214,7 @@ def initialize(seeds, monitoring, options = Options::Redacted.new)
 
       # Need to record start time prior to starting monitoring
       start_time = Time.now
```

Comment:
Is this start_time used for something now?

Reply:
It is used on line 244: `server.last_scan >= start_time`

The `last_scan` attribute of `Server` is public and explicitly documented with type `Time`. So, I am hesitant to change it.

Reply:
I think there need to be two last_scan fields, the new one being monotonic_last_scan or last_scan_monotonic (or, given the naming conventions I prefer, I would suggest last_scan_monotime as a contraction from last_scan_monotonic_time).

Last scan is used in two places: 

- The server selector uses it for staleness calculations. The value there is compared to the server-reported time and must therefore remain to be a wallclock time.
-Cluster uses it for waiting in the Client constructor until all servers have been scanned. The time there should be monotonic.

The monotonic last scan can be `@api private`.